app/src/store/contexts/socialNetworkProfileContext.jsx

```diff
@@ -25,9 +25,28 @@ export default function SocialNetworkProfileContextProvider({ children }) {
         setUserSocialNetworkProfile(temp_userProfile);
     }
 
+    async function followUser(selectedProfileId) {
+        return await callService(socialNetworkProfileservice, 'unfollowUser', [
+            selectedProfileId,
+            user._id,
+        ]);
+    }
+
+    async function unfollowUser(selectedProfileId) {
+        return await callService(socialNetworkProfileservice, 'followUser', [
+            selectedProfileId,
+            user._id,
+        ]);
+    }
+
     return (
         <SocialNetworkProfileContext.Provider
-            value={{ userSocialNetworkProfile, setUserSocialNetworkProfile }}
+            value={{
+                userSocialNetworkProfile,
+                setUserSocialNetworkProfile,
+                followUser,
+                unfollowUser,
+            }}
```

Comment:
Um lance que apontou no SonarCloud é a questão de re-render disso, então sugeriram utilizar o useMemo pra otimizar a performance do aplicativo. Avalia aí se faz sentido isso

```suggestion
const contextValue = useMemo(() => {
  return {
     userSocialNetworkProfile,
     setUserSocialNetworkProfile,
     followUser,
     unfollowUser,
  };
}, [userSocialNetworkProfile, followUser, unfollowUser]);

return (
  <SocialNetworkProfileContext.Provider value={contextValue}>
    {children}
  </SocialNetworkProfileContext.Provider>
);
```

Reply:
Mas caso não ocorra re-render desse estado, principalmente no userSocialNetworkProfile, pode ocorrer um caso onde ele vai poder mostrar os dados do usuário errado, já que o estado não vai ser limpo não?

Reply:
Não sei, não manjo tanto de useMemo, mas o chatGPT disse isso:

No, memoizing the context value with useMemo will not cause another user's data to be displayed.

The useMemo hook is used here to optimize performance by preventing unnecessary re-renders of consuming components. It does not affect the correctness of the context data.

The context data is still being updated based on the current user's ID passed to the getUserProfile function, and will not be affected by the memoization of the context value.

In fact, the memoization of the context value can help ensure that the correct data is displayed, because it can prevent the context provider from being re-rendered unnecessarily, which could potentially cause data inconsistencies.

So, memoizing the context value with useMemo is a safe and effective way to optimize your application's performance without compromising the accuracy of the displayed data.

Reply:
Sim sim, no caso não seria o useMemo em si que causaria a anomalia, mas algum lugar do código tentar acessa por exemplo se o valor é nulo antes de atualizar, vou verificar onde isso é utilizado porque era algo que já existia e caso tenha alguma lógica que possa causar anomalia vou corrigir então.